### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by vrx29**
This PR contains authentication logic for the ecom app. Private routing is also enable so you cannot access cart and wishlist without loggin in into the app. 
You can signup or use my credentials
Credentials - 
> id : vineet@gmail.com
> pass: vineet123

As there are multiple files you can review these files only.
:point_right: [AuthContext](https://github.com/vrx29/gimesto-store-ecom/pull/4/files#diff-f0b61a0825b8ed568eed5811cb2ebf19e5e50845327123432fcde8d229d72661)
:point_right: [authHandler Hook](https://github.com/vrx29/gimesto-store-ecom/pull/4/files#diff-434fa4a8cf7e06ff912e416d94c00897817378a341a989b27adb7403e0951366)
:point_right: [login page](https://github.com/vrx29/gimesto-store-ecom/pull/4/files#diff-da005ba6520f0f42b7e1052a931cea46e6703aceea8989e37fa93718ac033ad1)
:point_right: [signup page](https://github.com/vrx29/gimesto-store-ecom/pull/4/files#diff-f8e0433a71c75a9ec429c0f2aff047c9a183a5625e1e433f46e16f8716600f6e)
:point_right: [form reducer](https://github.com/vrx29/gimesto-store-ecom/pull/4/files#diff-2e97ddecad4042d88f7e3b00e1f7404e320d3fea129ed81e3f0ad8a74ea2bfac)

### CODE REVIEW

**Review by samyakshah3008 — APPROVED** [review]
Clean code and nice folder structure ✅
Good naming conventions  
Really liked how you provided preview description, helped for better review 

Good for approval

File: src/App.js [hunk 1]
```diff
@@ -28,10 +29,26 @@ function App() {
             </FilterProvider>
           }
         />
-        <Route path="/wishlist" element={<WishList />} />
-        <Route path="/cart" element={<Cart />} />
+        {/* <Route path="/cart" element={<Cart />} /> */}
```
Comment (Maianki) [inline src/App.js]: You can remove the commented code.

**Review by Maianki — APPROVED** [review]
Looks good to merge.
Things I liked about your code.
- Use of prop types 
- custom hooks and functions for abstraction.

File: src/App.js (hunk 1 above)
Comment (vrx29) [reply src/App.js]: sure will do this

**Pull request merged**